modules: openthread: undefined reference to `otThreadSetChildTimeout'

### Describe the bug

Commit eca4dee0464bfde86fd942e22f7a8d19dd76cd56 introduced a build error for the co-processor sample.

### Regression

- [x] This is a regression.

### Steps to reproduce

```shell
$ west build -p -b nrf52840dk/nrf52840 samples/net/openthread/coprocessor -T sample.openthread.coprocessor.rcp
```

### Relevant log output

```shell
/home/pdgendt/.local/opt/zephyr-sdk-0.17.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/openthread/libopenthread_utils.a(openthread.c.obj): in function `openthread_init':
/home/pdgendt/zephyrproject/zephyr/modules/openthread/openthread.c:359: undefined reference to `otChildSupervisionSetInterval'
/home/pdgendt/.local/opt/zephyr-sdk-0.17.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /home/pdgendt/zephyrproject/zephyr/modules/openthread/openthread.c:361: undefined reference to `otChildSupervisionSetCheckTimeout'
/home/pdgendt/.local/opt/zephyr-sdk-0.17.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /home/pdgendt/zephyrproject/zephyr/modules/openthread/openthread.c:363: undefined reference to `otThreadSetChildTimeout'
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

_No response_

### Additional Context

_No response_

--- a/modules/openthread/openthread.c
+++ b/modules/openthread/openthread.c
@@ -166,6 +166,22 @@ static void ot_joiner_start_handler(otError error, void *context)
 	}
 }
 
+static void ot_configure_instance(void)
+{
+#ifndef CONFIG_OPENTHREAD_COPROCESSOR_RCP
+	/* Configure Child Supervision and MLE Child timeouts. */
+	otChildSupervisionSetInterval(openthread_instance,
+				      CONFIG_OPENTHREAD_CHILD_SUPERVISION_INTERVAL);
+	otChildSupervisionSetCheckTimeout(openthread_instance,
+					  CONFIG_OPENTHREAD_CHILD_SUPERVISION_CHECK_TIMEOUT);
+	otThreadSetChildTimeout(openthread_instance, CONFIG_OPENTHREAD_MLE_CHILD_TIMEOUT);
+
+	if (IS_ENABLED(CONFIG_OPENTHREAD_ROUTER_SELECTION_JITTER_OVERRIDE)) {
+		otThreadSetRouterSelectionJitter(openthread_instance, OT_ROUTER_SELECTION_JITTER);
+	}
+#endif
+}
+
 static bool ot_setup_default_configuration(void)
 {
 	otExtendedPanId xpanid = {0};
@@ -355,17 +371,7 @@ int openthread_init(void)
 		}
 	}
 
-	/* Configure Child Supervision and MLE Child timeouts. */
-	otChildSupervisionSetInterval(openthread_instance,
-				      CONFIG_OPENTHREAD_CHILD_SUPERVISION_INTERVAL);
-	otChildSupervisionSetCheckTimeout(openthread_instance,
-					  CONFIG_OPENTHREAD_CHILD_SUPERVISION_CHECK_TIMEOUT);
-	otThreadSetChildTimeout(openthread_instance, CONFIG_OPENTHREAD_MLE_CHILD_TIMEOUT);
-
-	if (IS_ENABLED(CONFIG_OPENTHREAD_ROUTER_SELECTION_JITTER_OVERRIDE)) {
-		otThreadSetRouterSelectionJitter(openthread_instance, OT_ROUTER_SELECTION_JITTER);
-	}
-
+	ot_configure_instance();
 	openthread_mutex_unlock();
 
 	(void)k_work_submit_to_queue(&openthread_work_q, &openthread_work);

